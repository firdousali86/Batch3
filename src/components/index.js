import MyUserList from './MyUserList';
import UserProfile from './UserProfile';
import CustomComponent from './CustomComponent';
import MapControl from './MapControl';
import AppStateN from './AppStateN';
import ReactTestComponent from './ReactTestComponent';
import InputControl from './InputControl';

export {
  MyUserList,
  UserProfile,
  CustomComponent,
  MapControl,
  AppStateN,
  ReactTestComponent,
  InputControl,
};
