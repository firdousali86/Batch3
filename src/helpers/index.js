import PersistanceHelper from './PersistanceHelper';
import ApiHelper from './ApiHelper';
import LocationHelper from './LocationHelper';
import NotificationHelper from './NotificationHelper';
import CryptoHelper from './CryptoHelper';
import LocaleHelper from './LocaleHelper';
import LogHelper from './LogHelper';

export {
  PersistanceHelper,
  ApiHelper,
  LocationHelper,
  NotificationHelper,
  CryptoHelper,
  LocaleHelper,
  LogHelper,
};
