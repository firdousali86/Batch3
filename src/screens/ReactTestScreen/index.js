import {View, Text, Button} from 'react-native';
import React, {useState} from 'react';

export default function ReactTestScreen() {
  const [sometext, setsometext] = useState('test screen');

  return (
    <View>
      <Text testID="text_db">{sometext}</Text>
      <Button
        testID="button1"
        title="Press"
        onPress={() => {
          setsometext('something');
        }}
      />

      <Button
        testID="button2"
        title="Press"
        onPress={() => {
          setsometext('test screen');
        }}
      />
    </View>
  );
}
