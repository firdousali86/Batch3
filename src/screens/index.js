import TestFlexScreen from './TestFlexScreen';
import TestStatePropsScreen from './TestStatePropsScreen';
import Dashboard from './Dashboard';
import Login from './Login';
import TestPureComponent from './TestPureComponent';
import TestPropsScreen from './TestPropsScreen';
import TestReduxScreen from './TestReduxScreen';
import TestReduxClassScreen from './TestReduxClassScreen';
import ListScreen from './ListScreen';
import CartScreen from './CartScreen';
import ListApiScreen from './ListApiScreen';
import TestRefScreen from './TestRefScreen';
import TestReduxQuery from './TestReduxQuery';
import LocationScreen from './LocationScreen';
import TestImagePicker from './TestImagePicker';
import TestClassLifecycle from './TestClassLifecycle';
import TypeScriptScreen from './TypeScriptScreen';

export {
  TestFlexScreen,
  TestStatePropsScreen,
  Dashboard,
  Login,
  TestPureComponent,
  TestPropsScreen,
  TestReduxScreen,
  TestReduxClassScreen,
  ListScreen,
  CartScreen,
  ListApiScreen,
  TestRefScreen,
  TestReduxQuery,
  LocationScreen,
  TestImagePicker,
  TestClassLifecycle,
  TypeScriptScreen,
};
