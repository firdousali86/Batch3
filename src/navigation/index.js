import React, {useEffect, useState} from 'react';
import {Button} from 'react-native';
import {PersistanceHelper, NotificationHelper} from '../helpers';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {useNavigation} from '@react-navigation/native';
import {EventRegister} from 'react-native-event-listeners';
import {useDispatch, useSelector} from 'react-redux';
import {
  TestFlexScreen,
  TestStatePropsScreen,
  Dashboard,
  Login,
  TestPureComponent,
  TestPropsScreen,
  TestReduxScreen,
  TestReduxClassScreen,
  ListScreen,
  CartScreen,
  ListApiScreen,
  TestRefScreen,
  TestReduxQuery,
  LocationScreen,
  TestImagePicker,
  TestClassLifecycle,
  ReactTestScreen,
} from '../screens';

const Stack = createNativeStackNavigator();

const Navigator = props => {
  const user = useSelector(state => state.user);
  const navigation = useNavigation();

  useEffect(() => {
    // fetchUserEmail();

    // let event = EventRegister.addEventListener('LoginEvent', data => {
    //   fetchUserEmail();
    // });

    NotificationHelper.initializeFCM(
      rm => {
        console.log(rm);
      },
      rm => {
        console.log(rm);
        navigation.navigate('listScreen');
      },
    );
    NotificationHelper.checkFCMPermission();
    NotificationHelper.getToken();

    return () => {
      // EventRegister.removeEventListener(event);
    };
  }, []);

  // const fetchUserEmail = async () => {
  //   const userEmail = await PersistanceHelper.getValue('userEmail');

  //   if (userEmail && userEmail.length > 0) {
  //     setIsUserLoggedin(true);
  //   } else {
  //     setIsUserLoggedin(false);
  //   }
  // };

  const isUserLoggedIn = () => {
    return true;
    return user?.data?.id && user?.data?.id?.length > 15;
  };

  const getMainStack = () => {
    return (
      <Stack.Group>
        <Stack.Screen
          name="reactTestScreen"
          component={ReactTestScreen}
          options={{title: 'React test'}}
        />
        <Stack.Screen
          name="dashboard"
          component={Dashboard}
          options={{title: 'Dashboard'}}
        />
        <Stack.Screen
          name="testReduxScreen"
          component={TestReduxScreen}
          options={{title: 'Test Reduc func Component'}}
        />
        <Stack.Screen
          name="listApiScreen"
          component={ListApiScreen}
          options={{title: 'List Api Screen'}}
        />
        <Stack.Screen
          name="testClassLifecycle"
          component={TestClassLifecycle}
          options={{title: 'test lifecycle'}}
        />
        <Stack.Screen
          name="listScreen"
          component={ListScreen}
          options={{
            title: 'List Screen',
            headerRight: () => (
              <Button
                onPress={() => {
                  navigation.navigate('cartScreen');
                }}
                title="Cart"
                color="red"
              />
            ),
          }}
        />
        <Stack.Screen
          name="cartScreen"
          component={CartScreen}
          options={{title: 'Cart Screen'}}
        />

        <Stack.Screen
          name="testImagePicker"
          component={TestImagePicker}
          options={{title: 'Image picker'}}
        />
        <Stack.Screen
          name="locationScreen"
          component={LocationScreen}
          options={{title: 'Location Screen'}}
        />

        <Stack.Screen
          name="testReduxQuery"
          component={TestReduxQuery}
          options={{title: 'Test Redux Query Screen'}}
        />
        <Stack.Screen
          name="testRefScreen"
          component={TestRefScreen}
          options={{title: 'Test Ref Screen'}}
        />

        <Stack.Screen
          name="testReduxClassScreen"
          component={TestReduxClassScreen}
          options={{title: 'Test Redux class Component'}}
        />
        <Stack.Screen
          name="testPropsScreen"
          component={TestPropsScreen}
          options={{title: 'Test Props Component'}}
        />
        <Stack.Screen
          name="testPureComponent"
          component={TestPureComponent}
          options={{title: 'Test Pure Component'}}
        />
        <Stack.Screen
          name="testStateProps"
          component={TestStatePropsScreen}
          options={{title: 'Test State and Props'}}
        />
        <Stack.Screen
          name="testFlex"
          component={TestFlexScreen}
          options={{title: 'Testing Flex System'}}
        />
      </Stack.Group>
    );
  };

  const getAuthStack = () => {
    return (
      <Stack.Group>
        <Stack.Screen
          name="login"
          component={Login}
          options={{title: 'Login'}}
        />
      </Stack.Group>
    );
  };

  return (
    <Stack.Navigator>
      {isUserLoggedIn() ? getMainStack() : getAuthStack()}
    </Stack.Navigator>
  );
};

export default Navigator;
