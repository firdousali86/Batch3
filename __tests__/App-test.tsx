/**
 * @format
 */

import 'react-native';
import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
} from '@testing-library/react-native';
// import App from '../src/App';

import ReactTestScreen from '../src/screens/ReactTestScreen';
import {validatePassword} from '../src/utils';

// Note: import explicitly to use the types shiped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// it('renders correctly', () => {
//   renderer.create(<App />);
// });

describe('react test screen', () => {
  it('test screen check', () => {
    const {getByTestId} = render(<ReactTestScreen />);

    // const incrementButton = getByTestId('button_increment');
    const labeltext = getByTestId('text_db');

    expect(labeltext).toHaveTextContent('test screen');
  });

  it('test screen check after button press', () => {
    const {getByTestId} = render(<ReactTestScreen />);

    const labeltext = getByTestId('text_db');
    const button1 = getByTestId('button1');

    fireEvent.press(button1);

    expect(labeltext).not.toHaveTextContent('test screen');
    expect(labeltext).toHaveTextContent('something');
    expect(labeltext.props.children).toBe('something');
  });

  test('snapshot mismatch after state change', async () => {
    // Arrange
    const component = renderer.create(<ReactTestScreen />);
    const treeBeforeStateChange = component.toJSON();

    const {getByTestId} = render(<ReactTestScreen />);

    // Act
    fireEvent.press(getByTestId('button1'));

    // Assert
    await waitFor(() => {
      const treeAfterStateChange = component.toJSON();
      // Assert
      expect(treeAfterStateChange).toMatchSnapshot();
    });

    // Reset state
    fireEvent.press(getByTestId('button2'));

    await waitFor(() => {
      const treeAfterStateReset = component.toJSON();
      // Assert
      expect(treeAfterStateReset).toMatchSnapshot();
    });
  });
});

describe('password test suite', () => {
  it('check length', () => {
    const password = '123456';

    const result = validatePassword(password);

    expect(result).toBe(false);
  });

  it('check length', () => {
    const password = 'Abc123456&';

    const result = validatePassword(password);

    expect(result).toBe(true);
  });
});
