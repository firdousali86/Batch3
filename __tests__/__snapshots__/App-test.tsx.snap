// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react test screen snapshot mismatch after state change 1`] = `
<View>
  <Text
    testID="text_db"
  >
    test screen
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="button1"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="button2"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press
      </Text>
    </View>
  </View>
</View>
`;

exports[`react test screen snapshot mismatch after state change 2`] = `
<View>
  <Text
    testID="text_db"
  >
    test screen
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="button1"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="button2"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press
      </Text>
    </View>
  </View>
</View>
`;

exports[`react test screen snapshot testing 1`] = `
<View>
  <Text
    testID="text_db"
  >
    test screen
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="button1"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press
      </Text>
    </View>
  </View>
</View>
`;

exports[`react test screen snapshot testing 2`] = `
<View>
  <Text
    testID="text_db"
  >
    test screen
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="button1"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press
      </Text>
    </View>
  </View>
</View>
`;
